### ISSUE TITLE
GSoC categories and projects not displayed properly

### ISSUE THREAD

**Issue by mariobehling** [fossasia/labs.fossasia.org#183]
The Idea page at http://labs.fossasia.org/ideas.html gets issues (projects) directly from the github issue tracker at labs.fossasia.org. The idea is that all issues labeled "GSoC" appear with the header of the second tag e.g. Open Event. 

Some strings are hard-coded and Open Event shows up as "Open-Event". 

Issues labeled "Open Event Android" do not show up at all. Just an empty area show up for each issue.

![screenshot from 2017-01-31 18-29-46](https://cloud.githubusercontent.com/assets/1583873/22476720/b1182c88-e7e3-11e6-9a3d-1eafc94e05d8.png)

**Comment by Abhi2424shek** [comment]
There are two open event android sections @mariobehling is that needed.

**Comment by mariobehling** [comment]
Exactly, that is the bug.

**Comment by malikshubham827** [comment]
@mariobehling I wanted to work on this, but I think that the issue is now resolved as the site is now showing all the issues.

**Comment by mariobehling** [comment]
The issue is just circumvented by re-naming the projects and assigning them to one category. It is not solved.

**Comment by malikshubham827** [comment]
@mariobehling I used postman to make API request using github api v3 and found that the name of tags are same as they appear i.e.  `Open Event` is showing as `Open Event`. 

I could not find the `Open Event Android` label, so I created a temporary issue on a temporary repository but it is working fine; so no constraints on word size which I guessed initially.

I used the second label as the header and body as body in v3 api and here is my: [log file](https://drive.google.com/open?id=0B4JXPd5QX0mlTHpMZU5taDN4ZXM) .

Please tell me if there is anything I missed or mis-understood.
Here is my test-script in postman:
![fossasia-gsoc-label](https://cloud.githubusercontent.com/assets/13511528/23333142/7d14eb38-fbac-11e6-948d-483d53978e97.png)

Best
Shubham Malik

**Comment by mariobehling** [comment]
@malikshubham827 Please check out current live site at http://labs.fossasia.org/ideas.html

I added the tag "Open Event Android" and one issue into the category, but:
* the issue does not show up
* on the site it is spelled "Open-Event Android" instead of "Open Event Android"
* the "-" (dash) is also added in other headings - it should not be

**Comment by malikshubham827** [comment]
@mariobehling Thanks for your review. I tried but the github api is giving the correct infomation without the `-`.
Here is what I got from api using the above mentioned script in postman.
![open event android fossasia](https://cloud.githubusercontent.com/assets/13511528/23338635/1bbbebd6-fc36-11e6-8187-5bde1e2a192b.png)

If your are using github api it should return the same result.

**Comment by malikshubham827** [comment]
@mariobehling Yupp the site is showing the `-`. Let me check the source code. There must be an error in the back-end while sending the data.

**Comment by malikshubham827** [comment]
@mariobehling I think I kind of find out the root cause of all this. I have created the pull request [#212](https://github.com/fossasia/labs.fossasia.org/pull/212) regarding this.

TL;DR
Please refer to the [source code of ideas.html](https://github.com/fossasia/labs.fossasia.org/blob/gh-pages/ideas.html#L403)

(1) On the `line 403`, replace it by 
`tag = label["name"].split(' ').join('-')`
Reason: replace(" ", "-") in jquery converts only first space ` ` with `-`, so the result look like
`Open-Event Android` instead of `Open-Event-Android` for assigning it to the href attribute. It's because of this the text is not coming.

See how this looks now: 
![fossasia android](https://cloud.githubusercontent.com/assets/13511528/23339094/dcbb8684-fc40-11e6-878a-7c57d7bb2411.png)


(2) Now coming to the `-` in the name. It's used primarily for binding the link in the index in the starting to the description as shown above in pic. So I changed the `lines 415 and 419` which are using the same text content as the href attribute to the original one we got from github.



_Looking forward to work with you in GSoC , if selected._

**Comment by isuruAb** [comment]
this issue should be closed i think @mariobehling
